### ISSUE TITLE
find/create test data for the master bckg subtraction step

### ISSUE THREAD

**Issue by nden** [spacetelescope/jwst#3087]
13 sp

/grp/jwst/ssb/chartreuse/master_background_data

- [x] MIRI LRS: #3150

- [x] Nirspec FS: #3153

- [x] Nirspec IFU data: #3152

- [x] Nirspec MOS data: #3153

**Comment by nden** [comment]
Test data was created for NRS IFU, NRS FS and MIRI LRS. These the modes we are testing in this sprint.

**Comment by hbushouse** [comment]
I have created test data for the NIRSpec MOS mode as well. The files are available on central store in /grp/jwst/ssb/bushouse/jwst_data/NIRSpec/MSA/. The files are:

nrs_mos_sci_cal.fits - original science image, for comparison
nrs_mos_bkg_cal.fits - background signal centered at 20 DN/s with Poisson distribution
nrs_mos_bkg_x1d.fits - 1-D background spectrum for input to master background (single slit instance)
nrs_mos_sci+bkg_cal.fits - science + background, for input to master background step

**Comment by hbushouse** [comment]
By the way, nrs_mos_sci+bkg_cal.fits has about 11 extracted slitlets in it (MultiSlitModel with 11 slit instances), each of which covers a little different wavelength range. So this tests the ability to interpolate the master bkg spectrum into multiple different "spaces".

**Comment by jdavies-st** [comment]
If we are adding in a pedestal ± poisson noise as an artificial background into the `_rate` image, then running the resulting (sci+bkg) file through `Spec2Pipeline` to produce a `_cal` image, we also need to run the background image itself through `Spec2Pipeline`, including the `flat_field` step, so that the background image from which we produce an `_x1d` has `flat_field` applied too.  Same as the (sci+bkg).

I didn't realize this was important until talking to @philhodge this morning.  My mental model was incorrect.

**Comment by hbushouse** [comment]
That's what I've done for the MIRI LRS and NIRSpec MOS test data I've produced. I created a bkg rate image, and then ran it all the way through calspec2 to create the bkg x1d file to use as the user-supplied master background. I added that bkg rate image to an existing sci rate image, to form a sci+bkg rate image. That sci+bkg rate image was then run all the way through calspec2 to create the sci+bkg cal image to use as input to the master background step. So they've both been all the way through calspec2.

**Comment by philhodge** [comment]
And it looks as if you ran nrs_mos_sci_rate.fits through calspec2 as well.  We do need that, too.

**Comment by hbushouse** [comment]
Yes, the original sci (with no additional background) was also run all the way through calspec2, to produce a cal product, to use as comparison against the bkg-subtracted results. So all 3 versions were run through calspec2.

**Comment by hbushouse** [comment]
I've updated my "miri_lrs_bkg_x1d.fits" file to copy the values from the "NET" column to the "FLUX" column, so that the background spectral data are still in units of countrate. I also verified that the NIRSpec flux calibration factors are all set to 1.0 and hence the FLUX and NET columns of the "nrs_mos_bkg_x1d.fits" file that I created have identical data (the FLUX column is still DN/s, not flux). So that one's OK.
